week2/homework/exercises/1-blog-API/server.js

```diff
@@ -1,10 +1,94 @@
-const express = require('express')
+
+const express = require('express');
+const path = require('path');
+const fs = require('fs');
+
+
 const app = express();
- 
-
-// YOUR CODE GOES IN HERE
-app.get('/', function (req, res) {
-  res.send('Hello World')
-})
- 
-app.listen(3000)
\ No newline at end of file
+app.use(express.json());
+
+
+app.get(    '/blogs'        , (req,res) => readAllBlogs(req,res)    );
+app.get(    '/blogs/:title' , (req,res) => readSingleBlog(req,res)  );
+app.post(   '/blogs'        , (req,res) => creatBlog(req,res)       );
+app.put(    '/blogs/:title' , (req,res) => updateBlog(req,res)      );
+app.delete( '/blogs/:title' , (req,res) => deleteBlog(req,res)      );
```

Comment:
```suggestion
app.get(    '/blogs'        , readAllBlogs    );
app.get(    '/blogs/:title' , readSingleBlog  );
app.post(   '/blogs'        , creatBlog       );
app.put(    '/blogs/:title' , updateBlog      );
app.delete( '/blogs/:title' , deleteBlog      );
```
beautiful, I can almost cry. Here is a small tip, if your functions accept the same parameters there is no need for an arro function:

Reply:
> Awesome work @KhaledAlsaleh . Nice and well organized code. It would be a pleasure to work together with someone like you.

Thank you very much for the encouragement, your words make me so excited!  and it is an honor to learn from you.

Reply:
Yes I got it, thanks for feedback.